otel-tests/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+use opentelemetry::sdk::trace;
+use opentelemetry::sdk::Resource;
+use opentelemetry::KeyValue;
+use opentelemetry_otlp::WithExportConfig;
+use std::sync::Once;
+use tracing_subscriber::filter::LevelFilter;
+use tracing_subscriber::layer::SubscriberExt;
+use tracing_subscriber::util::SubscriberInitExt;
+use tracing_subscriber::EnvFilter;
+use tracing_subscriber::Layer;
+use tracing_subscriber::Registry;
+
+pub use futures;
+pub use opentelemetry;
+pub use otel_tests_macro::otel_test;
+pub use tokio;
+
+static INIT_OTLP_EXPORTER: Once = Once::new();
+
+pub fn init_tracing() {
+    INIT_OTLP_EXPORTER.call_once(|| {
+        let env = std::env::var("ITCHYSATS_TEST_INSTRUMENTATION").unwrap_or_default();
+        let telemetry = if env == "1" {
+            let cfg = trace::Config::default().with_resource(Resource::new([KeyValue::new(
+                "service.name",
+                "daemon-tests",
+            )]));
+
+            let tracer = opentelemetry_otlp::new_pipeline()
+                .tracing()
+                .with_trace_config(cfg)
+                .with_exporter(
+                    opentelemetry_otlp::new_exporter()
+                        .grpcio()
+                        .with_endpoint("localhost:4317"),
+                )
+                .install_simple()
+                .unwrap();
+
+            Some(tracing_opentelemetry::layer().with_tracer(tracer))
+        } else {
+            None
+        };
+
+        let filter = EnvFilter::from_default_env()
+            // apply warning level globally
+            .add_directive(LevelFilter::WARN.into())
+            // log traces from test itself
+            .add_directive("happy_path=debug".parse().unwrap())
```

Comment:
🤔 I guess this needs to be configurable if we wanna see test logs for anything other than `happy_path` tests.

Reply:
Yea, this (and the service name) still needs to be addressed. I'm open to suggestions on how to achieve configuration of these 2 aspects.

Reply:
as for the test log, we could agree that we use a "test" (or any other name) log target  by convention in all crates using `otel-tests`.

as for service name, can we somehow retrieve module name? (macro noob here 😅 )

Reply:
Macros transform input tokens to output tokens - they are not more special than that. So, if we can get the module name in normal rust code, then yes. I think this might be possible through panic info and location APIs?

Reply:
`module_path!()` might be a good enough approximation

Reply:
Done in latest commit